components/rmrk/Gallery/Holder/Holder.vue

```diff
@@ -345,37 +366,61 @@ export default class Holder extends mixins(ChainMixin, KeyboardEventsMixin) {
     return Object.values(itemRowMap)
   }
 
-  private generateHolderGroups(itemRowList): TableRow[] {
-    const holderGroups: Record<string, TableRow> = {}
-    itemRowList
-      .filter((item) => item.Holder !== '-')
-      .forEach((item) => {
-        const holder = item['Holder']
-        if (holderGroups[holder]) {
-          holderGroups[holder].Items.push(item)
-          holderGroups[holder]['Bought'] =
-            holderGroups[holder]['Bought'] + item['Bought']
-          holderGroups[holder]['Sale'] =
-            holderGroups[holder]['Sale'] + item['Sale']
-        } else {
-          holderGroups[holder] = {
-            ...item,
-            Items: [item],
-          }
+  private getGroupNameFromRow(item: TableRow): string {
+    if (this.groupKey === 'Holder') {
+      return item['Holder']
+    } else if (this.groupKey === 'CollectionId') {
+      return item['Item']['collection']['id']
+    }
+    return item['Holder']
+  }
+
+  private getCustomRowFilter(): (item: TableRow) => boolean {
+    if (this.groupKey === 'Holder') {
+      return (item) => item.Holder !== '-'
+    } else if (this.groupKey === 'CollectionId') {
+      return (item) => item.Holder === this.$route.params.id
+    }
+    return () => true
```

Comment:
another small refacto
```suggestion
    return this.groupKey !== 'Holder'
			? (item) => item.Holder === this.$route.params.id
			: (item) => item.Holder !== '-'
```

Reply:
It seems that it don't need to change, because we may add more group type in the future, not just 'Holder' and 'CollectionId'.